vita3k/shader/src/spirv_recompiler.cpp

```diff
@@ -1374,14 +1414,47 @@ static spv::Function *make_vert_finalize_function(spv::Builder &b, const SpirvSh
 
                 cond_builder.makeBeginElse();
 
+                // Apply the viewport flip
                 if (translation_state.render_info_id != spv::NoResult) {
```

Comment:
I'm hazy on memory but the all intentions of this flipping machinary was because opengl didn't have enough tools to do this without hardcoding. With vulkan, it should be possible to use negative viewport size to emulate this cleanly.

Reply:
Oh indeed, I didn't know you could specify a negative width and height in a viewport, this will be much cleaner.

Reply:
With the latest update, a negative viewport height is used instead of the viewport flip in the shader.